autogpt_platform/frontend/src/app/(platform)/auth/auth-code-error/page.tsx

```diff
@@ -23,15 +28,105 @@ export default function AuthErrorPage() {
   }, []);
 
   if (!errorType && !errorCode && !errorDescription) {
-    return <div>Loading...</div>;
+    return (
+      <div className="flex h-screen items-center justify-center">
+        <Text variant="body">Loading...</Text>
+      </div>
+    );
   }
 
+  // Check if this is a waitlist/not allowed error
+  const isWaitlistError =
+    errorCode === "403" ||
+    errorDescription?.toLowerCase().includes("not allowed") ||
+    errorDescription?.toLowerCase().includes("waitlist") ||
+    errorDescription?.toLowerCase().includes("allowlist");
+
+  if (isWaitlistError) {
+    return (
+      <div className="flex h-screen items-center justify-center">
+        <Card className="w-full max-w-md p-8">
+          <div className="flex flex-col items-center gap-6">
+            <Text variant="h3">Join the Waitlist</Text>
+            <div className="flex flex-col gap-4 text-center">
+              <Text variant="body">
+                AutoGPT Platform is currently in closed beta. Your email address
+                isn&apos;t on our current allowlist for early access.
+              </Text>
+              <Text variant="small" className="text-muted-foreground">
+                Join our waitlist to get notified when we open up access!
+              </Text>
+            </div>
+            <div className="flex gap-3">
+              <Button
+                variant="secondary"
+                onClick={() => {
+                  window.open("https://agpt.co/waitlist", "_blank");
+                }}
+              >
+                Join Waitlist
+              </Button>
+              <Button variant="primary" onClick={() => router.push("/login")}>
+                Back to Login
+              </Button>
+            </div>
+            <div className="flex flex-col gap-2">
+              <Text
+                variant="small"
+                className="text-center text-muted-foreground"
+              >
+                Already signed up for the waitlist? Make sure you&apos;re using
+                the exact same email address you used when signing up.
+              </Text>
+              <Text
+                variant="small"
+                className="text-center text-muted-foreground"
+              >
+                If you&apos;re not sure which email you used or need help,{" "}
+                <a
+                  href="https://discord.gg/autogpt"
+                  target="_blank"
+                  rel="noopener noreferrer"
+                  className="underline hover:text-foreground"
+                >
+                  reach out on Discord
+                </a>
+              </Text>
+            </div>
+          </div>
```

Comment:
This is 99% the same as what's in `EmailNotAllowedModal`. Can you break it out into a re-usable component and use that here?